snakemake/settings/types.py

```
@@ -312,8 +315,11 @@ def _get_greediness(self):
             return self.greediness
 
     def _check(self):
-        if not (0 < self.greedyness <= 1.0):
-            raise ApiError("greediness must be >0 and <=1")
+        if not (0 <= self.greediness <= 1.0):
+            raise ApiError("greediness must be >=0 and <=1")
+        if self.subsample:
+            if not isinstance(self.subsample, int) or self.subsample < 1:
```

Ruff

Use a single `if` statement instead of nested `if` statements

(SIM102)